### DISCUSSION TITLE
Custom Authentication Function

### DISCUSSION THREAD

**Discussion by ygarg465** [electric-sql/electric#879] [Feature requests]
Can we expect an authentication mechanism that executes a custom serverless function to verify the request, which can then perform certain actions to accept or reject the request as it gives more flexibility instead of having the Electric server authenticate it using a JWT key?

We expect ElectricSQL to be used as a drop-in DB inside of an already existing project which then needs to be adjusted to use ElectricSQL-based authentication.

Currently, in my project, I use dynamic JWT keys for every user to authenticate their request but in the case of Electric I can't, so I have to adjust or create another implementation of authentication in my backend for that.

For reference, what I am talking about have a look at this [MongoDB Custom Authentication Function](https://www.mongodb.com/docs/atlas/app-services/authentication/custom-function/)

**Comment by alco** [comment]
Relevant internal issue: VAX-282.

**Comment by KyleAMathews** [comment]
👋 we've been working the last month on a rebuild of the Electric server over at a temporary repo https://github.com/electric-sql/electric-next/

You can read more about why we made the decision at https://next.electric-sql.com/about

We're really excited about all the new possibilities the new server brings and we hope you'll check it out soon and give us your feedback.

We're now moving the temporary repo back here. As part of that migration we're closing all the old issues, PRs, and discussions. We really appreciate you taking the time to start this discussion! If it's still relevant in the new system, please reopen.
